ReasoningEngine/Command.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using System.Threading.Tasks;
+using ReasoningEngine.GraphFileHandling;
+
+namespace ReasoningEngine.Communication
+{
+    public class CommandProcessor
+    {
+        private readonly GraphFileManager graphFileManager;
+
+        public CommandProcessor(GraphFileManager graphFileManager)
+        {
+            this.graphFileManager = graphFileManager;
+        }
+
+        /// <summary>
+        /// Displays the command menu and processes user input.
+        /// </summary>
+        public void ShowCommandMenu()
+        {
+            while (true)
+            {
+                Console.WriteLine("\nCommand Menu:");
+                Console.WriteLine("1. Node Query");
+                Console.WriteLine("2. Edge Query");
+                Console.WriteLine("3. Add Node");
+                Console.WriteLine("4. Delete Node");
+                Console.WriteLine("5. Edit Node");
+                Console.WriteLine("6. Add Edge");
+                Console.WriteLine("7. Delete Edge");
+                Console.WriteLine("8. Edit Edge");
+                Console.WriteLine("9. Exit");
+                Console.Write("Enter your choice: ");

```

Comment:
EDIT:

I'm going to change the length of the 4-character string to be a 6-character string in a future update, so use 6 characters instead of 4 please.